core/rawdb/accessors_metadata.go

```diff
@@ -84,18 +84,104 @@ func WriteChainConfig(db ethdb.KeyValueWriter, hash common.Hash, cfg *params.Cha
 	}
 }
 
-// ReadGenesisStateSpec retrieves the genesis state specification based on the
-// given genesis (block-)hash.
-func ReadGenesisStateSpec(db ethdb.KeyValueReader, blockhash common.Hash) []byte {
-	data, _ := db.Get(genesisStateSpecKey(blockhash))
-	return data
-}
+const (
+	maxChunkSize = 4 * 1024 * 1024 // 4MB
+	chunkPrefix  = 0x00            // Special prefix indicating chunked data
+)
 
 // WriteGenesisStateSpec writes the genesis state specification into the disk.
+// If the data is larger than 4MB, it will be split into chunks.
```

Comment:
better to re-implement a new `WriteGenesisStateSpecLarge` method, separating the newly introduced logic from the existing logic to reduce mixture with the original logic.